package pb;

import "CommomDef.proto";




message GS2C_StageLevel
{
    optional uint32 level_id                               =1;
    optional uint32 fight_count		                       =2;
    optional bool   is_clearance                           =3;
}

//@SMSG_ALL_STAGE_INFO
message GS2C_AllStageInfo
{
    repeated GS2C_StageLevel stage_level_infos = 1;
}

//@CMSG_BATTLE_DATA
message C2GS_BattleData
{
    required uint32 stage_level_id    = 1;
	optional int32  pre_battle_id     = 2; // 用于验证。
}
//关卡奖励结果
message GS2C_Reward
{
    repeated PairValue curency_type_count = 1;
    repeated PairValue item_id_count      = 2;
    optional uint32    exp                = 3;    
}

//@SMSG_BATTLE_RESULT
message GS2C_BattleResult
{
    optional uint32          err_code          = 1;
	optional bool	         is_battle_win	   = 2;
    optional GS2C_StageLevel stage_level_info  = 3;
    optional GS2C_Reward     battle_reward     = 4;
}


